### PR TITLE
Release 0.7.1

### PR DESCRIPTION
**Pull request by nicoddemus**
OK now we are in business.

Had to call `git commit` directly in the commit script because apparently gitpython bypasses the local configuration (don't have time right now to look for another solution).

Other than that, I changed #131 and #66 from `feature` to `trivial`, as those are more internal changes than user-facing features.

### CODE REVIEW

**Comment by tgoodlet** [comment]
So this is where we click merge and everything just happens?

**Comment by RonnyPfannschmidt** [comment]
@tgoodlet it still needs a tag as well

**Comment by tgoodlet** [comment]
Version conflict in the [benchmark tests](https://travis-ci.org/pytest-dev/pluggy/jobs/408922017#L489)?

**Comment by nicoddemus** [comment]
Bummer:

```
pkg_resources.ContextualVersionConflict: (pluggy 0.7.0 (/home/travis/build/pytest-dev/pluggy/.tox/benchmark/lib/python3.6/site-packages), Requirement.parse('pluggy<0.7,>=0.5'), {'pytest'})
```

pytest requires `pluggy<0.7`... not sure why this didn't blow up with the other jobs. 🤔

**Comment by tgoodlet** [comment]
@nicoddemus maybe a race with the PyPi upload? Benchmark tests are the last job?

**Comment by nicoddemus** [comment]
No, it is a conflict with the pluggy being built...

It is weird because the other jobs installed `pytest` and `pluggy` even with the restriction in place:

[py27release](https://travis-ci.org/pytest-dev/pluggy/jobs/408922007)

```
============================= test session starts ==============================
platform linux2 -- Python 2.7.14, pytest-3.6.3, py-1.5.4, pluggy-0.7.0
```

And also it seems we have a hidden bug in pluggy:

```
  File "/home/travis/build/pytest-dev/pluggy/.tox/benchmark/lib/python3.6/site-packages/pluggy/manager.py", line 258, in load_setuptools_entrypoints
    "Plugin %r could not be loaded: %s!" % (ep.name, e))
TypeError: __init__() missing 1 required positional argument: 'message'
```

**Comment by nicoddemus** [comment]
I propose that we skip the `benchmark` job when creating the package... as #164 shows, it is failing when loading the `benchmark` entry point, but is not the fault of the benchmark specifically because it does not pin `pluggy`. I'm guessing it is a problem with `pip`'s dependency solver.

**Comment by nicoddemus** [comment]
Another solution is to release a new pytest version with `pluggy<0.8` pinnings.

**Comment by nicoddemus** [comment]
Started pytest 3.6.4 which loosens the pin a bit to `pluggy<0.8`: https://github.com/pytest-dev/pytest/pull/3728

Seems like the easiest solution for now because we won't need to change anything on `pluggy`. Still, it is an interesting problem because it doesn't seem to exist a direct solution to this, they are all compromises.

**Comment by nicoddemus** [comment]
With the new pytest released which pins to `pluggy<0.8` this should now work. I've also cherry-picked the fix in #164 to here.

I had to manually update the changelog to mark `0.7.0` as never released to PyPI, let me know if you guys thinks that is fine.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Pluggy 0.7.0 (2018-07-26)
+pluggy 0.7.1 (2018-07-28)
```
Comment (RonnyPfannschmidt) [inline CHANGELOG.rst]: a p is missinghere

File: CHANGELOG.rst [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
 =========================
 
 Deprecations and Removals
```
Comment (RonnyPfannschmidt) [inline CHANGELOG.rst]: dito

**Comment by RonnyPfannschmidt** [comment]
i beleive the templates are broken for multiple projects

**Comment by nicoddemus** [comment]
Actually it is a problem with the GH extension we use, see my other replies 😁

**Comment by tgoodlet** [comment]
So we're good to go here now. Nice job on the bug fix btw @nicoddemus.

**Comment by nicoddemus** [comment]
Yep we just need to push a tag here. Want to do the honors here @tgoodlet? I'm away from the computer now

**Comment by tgoodlet** [comment]
@nicoddemus tag is `0.7.1` yah?

**Comment by nicoddemus** [comment]
Yes :)

Em sáb, 28 de jul de 2018 17:12, goodboy <notifications@github.com>
escreveu:

**Comment by tgoodlet** [comment]
Weird, I totally didn't know you can *just* push a tag without the branch.
Anyway, think I did it?

**Comment by tgoodlet** [comment]
Yup. [We're live](https://pypi.org/project/pluggy/) :+1:

**Comment by tgoodlet** [comment]
I guess the only thing remaining is the anaconda package?

**Comment by nicoddemus** [comment]
That happens automatically :)

**Comment by nicoddemus** [comment]
> I guess the only thing remaining is the anaconda package?

@tgoodlet FYI https://github.com/regro-cf-autotick-bot picks up releases to PyPI and automatically opens a PR to update the conda package. 😉 

https://github.com/conda-forge/pluggy-feedstock/pull/6

**Pull request merged**